cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -264,15 +264,35 @@ class ARROW_EXPORT AzureFileSystem : public FileSystem {
 
   Status CreateDir(const std::string& path, bool recursive) override;
 
+  /// \brief Delete a directory and its contents recursively.
+  ///
+  /// Atomicity is guaranteed only on Hierarchical Namespace Storage accounts.
   Status DeleteDir(const std::string& path) override;
 
+  /// \brief Non-atomically deletes the contents of a directory.
+  ///
+  /// This function can return a bad Status after only partially deleting the
+  /// contents of the directory.
   Status DeleteDirContents(const std::string& path, bool missing_dir_ok) override;
 
+  /// \brief Deletion of all the containers in the storage account (not
+  /// implemented for safety reasons).
+  ///
+  /// \return Status::NotImplemented
   Status DeleteRootDirContents() override;
 
+  /// \brief Deletes a file.
+  ///
+  /// Supported on both flat namespace and Hierarchical Namespace storage
+  /// accounts. A check is made to guarantee the parent directory doesn't
+  /// disappear after the blob is deleted and while this operation is running,
```

Comment:
Is this referring to flat namespace accounts only? I can't imagine a hierarchical namespace directory disappearing when a file is deleted.

Reply:
Re-wording suggestion? My intention is being concise, but yeah, none of this applies to HNS accounts.

Reply:
Less concise but I would go for 
```suggestion
  /// accounts. On flat namespace storage accounts a check is made to guarantee the parent directory doesn't
  /// disappear after the blob is deleted and while this operation is running,
```
I don't feel strongly